lib/discrete_maths/logic_operations.py

```diff
@@ -0,0 +1,178 @@
+# Classical logic operations library
+
+from lib.parsers import shunting_yard
+
+# Operator Precedence Dictionary - = == iff, > == implies, v == or, ^ == and, ~ == not
+OP_PRECIDENCE = {"=": 2, ">": 3, "v": 4, "^": 5, "~": 6}
```

Comment:
There's a typo here. It's `precedence`. Additionally it would be better if it was called `OPERATION_PRECEDENCES` or `OPERATION_BY_PRECEDENCES` (the use of `by` is commonly used in hashes to describe what it is)